### ISSUE TITLE
Add toyaml, fromyaml context methods

### ISSUE THREAD

**Issue by clrcrl** [fishtown-analytics/dbt#1911]
### Describe the feature
Make `toyaml` and `fromyaml` methods.
`toyaml` should have an optional argument, `sort_keys=False`.
Similar to `tojson`, `fromjson`  methods.
I'd use this when using a macro to generate schema.yml files.

### Describe alternatives you've considered
[Here](https://github.com/fishtown-analytics/dbt-codegen/blob/master/macros/generate_source.sql#L25) is how I currently do it:
```
{% set sources_yaml=[] %}

{% do sources_yaml.append('version: 2') %}
{% do sources_yaml.append('') %}
{% do sources_yaml.append('sources:') %}
{% do sources_yaml.append('  - name: ' ~ schema_name | lower) %}
...
```

**Comment by clrcrl** [comment]
The default yaml dumper produces yaml that doesn't match our [YAML style guide](https://github.com/fishtown-analytics/corp/blob/master/dbt_coding_conventions.md#yaml-style-guide).

Nice to have: Implement a default Dumper that dumps yaml the way we like.

**Comment by beckjake** [comment]
Fixed in #2036
